Asterix/optics/propagation_functions.py

```diff
@@ -165,17 +225,50 @@ def mft(image,
             norm0 = np.sqrt(nbresx * nbresy / dim_input_x / dim_input_y / dim_output_x / dim_output_y)
         sign_exponential = 1
 
-    AA = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(uu0, xx0))
-    BB = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(xx1, uu1))
-    result = norm0 * np.matmul(np.matmul(AA, image), BB)
+    AA = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(uu0, xx0)).astype('complex64')
+    BB = np.exp(sign_exponential * 1j * 2 * np.pi * np.outer(xx1, uu1)).astype('complex64')
 
-    return result
+    if returnAABB:
+        return AA, BB, norm0
 
+    return mat_mult_mft(image.astype('complex64'), AA, BB, norm0)
 
-def prop_fresnel(pup, lam, z, rad, prad, retscale=0):
+
+def mat_mult_mft(image, AA, BB, norm0):
+    """Perform Matrix multiplication for MFT. It is be done separatly in to
+    allowed to be sped up by GPU (maybe). I tried using numba for this function
+    to save time but no improvements.
+
+    complex64 type is not mandatory but helps greatly speed up the code. I
+    realized that previously (before 2022-10-11) image was float64/complex64, while AA and BB where complex128.
+    Therefore the results of the MFT was artificially extended to complex128.
+
+    AUTHOR : Johan Mazoyer
+        2022-10-11 Creation from MFT
+
+    Parameters
+    ----------
+    image : 2D numpy array (complex64)
+            Entrance image
+    AA: 2D numpy array (complex64)
+            Matrix multiplied in norm0 * ((AA @ image) @ BB). This parameter is only used if only_mat_mult = True
+    BB: 2D numpy array (complex64)
+            Matrix multiplied in norm0 * ((AA @ image) @ BB). This parameter is only used if only_mat_mult = True
+    norm0: float, default
+            Normalization value in matrix multiplication norm0 * ((AA @ image) @ BB).
+            This parameter is only used if only_mat_mult = True
```

Comment:
The "This parameter is only used..." part has nothing to do with this function and should be deleted. It shows up in the parameter docstring of the `mft()` function anyway, where it should be 👍

Reply:
ok